src/AnalyticsEventReporter.ts

```diff
@@ -34,10 +34,31 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
   with(payload: EventPayload): AnalyticsEventService {
     const currentPayload =
       this.payload === undefined ? payload : merge(this.payload, payload);
+    if (this.config.debug) {
+      console.log('[DEBUG] AnalyticsConfig Object:\n', this.config);
+      console.log(
+        '[DEBUG] Payload following call to `with`:\n',
+        currentPayload
+      );
+    }
     return new AnalyticsEventReporter(this.config, currentPayload);
   }
 
   public async report(newPayload?: EventPayload): Promise<string> {
+    if (this.config.debug) {
+      console.log(
+        `[DEBUG] Merging the following payloads:
+
+        Original Payload from call to \`with\`:
+
+        ${this.payload ? JSON.stringify(this.payload) : '{}'}
```

Comment:
Is there a reason why this is formatted differently than the other logs? My understanding was to keep this as one line, but can't just just do two console logs?